File: ProjectAndroidStudio/MySQLdatabase/app/src/main/java/com/komputerkit/mysqldatabase/BarangAdapter.java
package com.komputerkit.mysqldatabase;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class BarangAdapter extends RecyclerView.Adapter<BarangAdapter.BarangViewHolder> {
    private ArrayList<Barang> listBarang;

    public BarangAdapter(ArrayList<Barang> listBarang) {
        this.listBarang = listBarang;
    }

    @NonNull
    @Override
    public BarangViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_barang, parent, false);
        return new BarangViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BarangViewHolder holder, int position) {
        Barang barang = listBarang.get(position);
        holder.tvNamaBarang.setText(barang.getNamaBarang());
        holder.tvStok.setText("Stok: " + barang.getStok());
        holder.tvHarga.setText("Harga: " + barang.getHarga());
    }

    @Override
    public int getItemCount() {
        return listBarang.size();
    }

    public void setListBarang(ArrayList<Barang> listBarang) {
        this.listBarang = listBarang;
        notifyDataSetChanged();
    }

    public static class BarangViewHolder extends RecyclerView.ViewHolder {
        TextView tvNamaBarang, tvStok, tvHarga;
        public BarangViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNamaBarang = itemView.findViewById(R.id.tvNamaBarang);
            tvStok = itemView.findViewById(R.id.tvStok);
            tvHarga = itemView.findViewById(R.id.tvHarga);
        }
    }
}

File: ProjectAndroidStudio/MySQLdatabase/app/src/main/java/com/komputerkit/mysqldatabase/BarangAdapter.kt
package com.komputerkit.mysqldatabase

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

// Adapter RecyclerView untuk daftar barang
class BarangAdapter(private var barangList: List<Barang>) : RecyclerView.Adapter<BarangAdapter.BarangViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BarangViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_barang, parent, false)
        return BarangViewHolder(view)
    }

    override fun onBindViewHolder(holder: BarangViewHolder, position: Int) {
        val barang = barangList[position]
        holder.tvNamaBarang.text = barang.namaBarang
        holder.tvStok.text = "Stok: ${barang.stok}"
        holder.tvHarga.text = "Harga: ${barang.harga}"
    }

    override fun getItemCount(): Int = barangList.size

    // Untuk update data barang di adapter
    fun setBarangList(newList: List<Barang>) {
        barangList = newList
        notifyDataSetChanged()
    }

    // ViewHolder untuk item barang
    class BarangViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvNamaBarang: TextView = itemView.findViewById(R.id.tvNamaBarang)
        val tvStok: TextView = itemView.findViewById(R.id.tvStok)
        val tvHarga: TextView = itemView.findViewById(R.id.tvHarga)
    }
}
